### ISSUE TITLE
How to reproduce the data of LPIPS distance?

### ISSUE THREAD

**Issue by WorkingCC** [junyanz/BicycleGAN#26]
Your work gets surprising results and I expect to reproduce the data of LPIPS distance that you list in Figure6. Given one input image, you sample 19 outputs. For every input(maps), do you calculate the LPIPS distance between the given image(maps) and corresponding 19 samples(satellite) ? After that, you sum those 19 groups of data and have a average? Is it the same to other 99 input images in your experiment ? I'm confused about this and looking forward to your reply, thank you!

**Comment by richzhang** [comment]
For each input image, we produce 20 outputs and compute the distance between each pair of _outputs_, and take the average. We do this across input images, and take the average.

**Comment by WorkingCC** [comment]
@richzhang ,thank you, I get the point but still have some questions. Do you compute LPIPS distance between the input image(maps) and the sampled output, or between the corresponding real satellite image and the sampled output?

**Comment by richzhang** [comment]
Neither - between two random outputs

**Comment by WorkingCC** [comment]
For each input image, you sample two random codes Z and generated two random outputs  at one time. And you do it up to 19 times, i.e., for each input image, you finally get 19 pairs of outputs. After it, you compute LPIPS distance  between 19 pairs of outputs and take the average. In your experiment, you do the same operation on 100 inputs and  get 1900 pairs of outputs in total, which are used to compute average LPIPS distance.  Is it right? @richzhang

**Comment by richzhang** [comment]
Yup! @WorkingCC

**Comment by richzhang** [comment]
Yes, they are randomly selected real images. It serves as a "ceiling" -- the results that an algorithm generates given a single A should not be greater than the variation given random ground truth images B.

**Comment by ElternalEnVy** [comment]
> For each input image, you sample two random codes Z and generated two random outputs at one time. And you do it up to 19 times, i.e., for each input image, you finally get 19 pairs of outputs. After it, you compute LPIPS distance between 19 pairs of outputs and take the average. In your experiment, you do the same operation on 100 inputs and get 1900 pairs of outputs in total, which are used to compute average LPIPS distance. Is it right? @richzhang

I think this guy may have some wrong understandings. The right way the collaborator explains is that first you sample 20 images for each real input image, and then you average the distances over all possible pairs of these 20 images which is C_20^2=1900. Finally, this process is repeated over all test images and you get the final score which means if you have n test images, there are 1900 \times n pairs are averaged.
